src/Commands/Diagnose.php

```diff
@@ -21,207 +19,122 @@
+        $this->logger->notice('{label}: {info} {status}', $setting);
       }
       else {
-        $this->logger->notice('We got Mbeans stats ✅');
+        $this->logger->notice('{label}: {info}', $setting);
       }
     }
-    catch (\Exception $e) {
-      \Kint::dump($e);
-      $this->logger->emergency("There's a problem somewhere...");
-      exit(1);
-    }
-    catch (\Throwable $t) {
-      \Kint::dump($t);
-      $this->logger->emergency("There's a problem somewhere...");
-      exit(1);
-    }
-    $this->logger->notice(
-          "If there's an issue with the connection, it would have shown up here. You should be good to go!"
-      );
   }
 
   /**
-   * Pings the Solr host.
-   *
-   * @usage search-api-pantheon:ping
-   *   Ping the solr server.
-   *
-   * @command search-api-pantheon:ping
-   * @aliases sapp
-   *
-   * @return \Solarium\Core\Query\Result\ResultInterface|\Solarium\QueryType\Ping\Result|void
-   *   The result.
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   * @SuppressWarnings(PHPMD.NPathComplexity)
    */
-  public function pingSolrHost() {
-    try {
-      $ping = $this->solr->createPing();
-      return $this->solr->ping($ping);
+  public function verifyYamlFiles(): void {
+    $pantheon_yml_contents = '';
+    if (file_exists($this->drupalRoot . '/pantheon.yml')) {
+      $pantheon_yml_contents = file_get_contents($this->drupalRoot . '/pantheon.yml');
     }
-    catch (\Exception $e) {
-      exit($e->getMessage());
+    elseif (file_exists($this->drupalRoot . '/../pantheon.yml')) {
+      $pantheon_yml_contents = file_get_contents($this->drupalRoot . '/../pantheon.yml');
     }
-    catch (\Throwable $t) {
-      exit($t->getMessage());
+    $pantheon_upstream_yml_contents = '';
+    if (file_exists($this->drupalRoot . '/pantheon.upstream.yml')) {
+      $pantheon_upstream_yml_contents = file_get_contents($this->drupalRoot . '/pantheon.upstream.yml');
+    }
+    elseif (file_exists($this->drupalRoot . '/../pantheon.upstream.yml')) {
+      $pantheon_upstream_yml_contents = file_get_contents($this->drupalRoot . '/../pantheon.upstream.yml');
+    }
+    if (!$pantheon_yml_contents && !$pantheon_upstream_yml_contents) {
+      throw new \Exception('Unable to find pantheon.yml or pantheon.upstream.yml');
+    }
+    $pantheon_yml = Yaml::parse($pantheon_yml_contents);
+    $pantheon_upstream_yml = Yaml::parse($pantheon_upstream_yml_contents);
+    if (empty($pantheon_yml['search']['version'])) {
+      // Merge from pantheon_upstream as fallback.
+      if (!empty($pantheon_upstream_yml['search']['version'])) {
+        $pantheon_yml['search']['version'] = $pantheon_upstream_yml['search']['version'];
+      }
+    }
+    if (empty($pantheon_yml['search']['version'])) {
+      // If still empty, throw an exception.
+      throw new \Exception('Unable to find search.version in pantheon.yml or pantheon.upstream.yml');
+    }
```

Comment:
Today, the default solr version when solr is enabled with nothing in p.yml is solr 3. In the future, that will change, and we will enable solr 8 by default. Should we be concerned that this heathcheck will fail once that happens? Or leave that for a future version of search_api_pantheon?